src/snowflake/cli/plugins/nativeapp/lsp_commands.py

```diff
@@ -0,0 +1,88 @@
+from typing import TypedDict
+
+from pygls.server import LanguageServer
+from snowflake.cli.api.output.types import MessageResult
+from snowflake.cli.api.project.definition_manager import DefinitionManager
+from snowflake.cli.plugins.nativeapp.init import nativeapp_init
+from snowflake.cli.plugins.nativeapp.manager import NativeAppManager
+from snowflake.cli.plugins.rpc.manager import LSPPluginContext
+from snowflake.snowpark.dataframe import DataFrame
+
+
+def lsp_plugin(name, version, capabilities):
+    def _decorator(func):
+        func.is_lsp_plugin = True
+        func.lsp_plugin_name = name
+        func.lsp_plugin_version = version
+        func.lsp_plugin_capabilities = capabilities
+        return func
+
+    return _decorator
+
+
+class ConnectionParams(TypedDict):
+    sessionToken: str
+    masterToken: str
+    account: str
+    host: str
+    port: str
+    protocol: str
```

Comment:
How different this is from `_ConnectionContext`?

Reply:
The host/port/protocol part is different, since the Python connector allows for providing those values and I wanted to make sure it'd work for those situations too.